exporter/splunkhecexporter/client_test.go

```diff
@@ -994,14 +960,36 @@ func TestSubLogs(t *testing.T) {
 	assert.Equal(t, "1_1_9", got.ResourceLogs().At(2).InstrumentationLibraryLogs().At(0).Logs().At(9).Name())
 }
 
-// validateCompressedEqual validates that GZipped `got` contains `expected` string
-func validateCompressedEqual(t *testing.T, expected string, got []byte) {
+// validateCompressedEqual validates that GZipped `got` contains `expected` strings
+func validateCompressedContains(t *testing.T, expected []string, got []byte) {
 	z, err := gzip.NewReader(bytes.NewReader(got))
 	require.NoError(t, err)
 	defer z.Close()
 
 	p, err := ioutil.ReadAll(z)
 	require.NoError(t, err)
 
-	assert.Equal(t, expected, string(p))
+	for _, e := range expected {
+		assert.Contains(t, string(p), e)
+	}
+
+}
+
+func BenchmarkPushLogRecords(b *testing.B) {
+	logs := createLogData(1, 1, 1)
+	c := client{
+		url: &url.URL{Scheme: "http", Host: "splunk"},
+		zippers: sync.Pool{New: func() interface{} {
+			return gzip.NewWriter(nil)
+		}},
+		config: NewFactory().CreateDefaultConfig().(*Config),
+		logger: zap.NewNop(),
+	}
+	sender := func(ctx context.Context, buffer *bytes.Buffer, headers map[string]string) error {
+		return nil
+	}
+	state := makeBlankBufferState(4096)
+	for n := 0; n < b.N; n++ {
+		_, _ = c.pushLogRecords(context.Background(), logs.ResourceLogs(), &state, map[string]string{}, sender)
```

Comment:
There is a potential the compiler can optimise these values if they are not being assigned to something.

https://dave.cheney.net/high-performance-go-workshop/dotgo-paris.html#fixing_the_benchmark

Reply:
@atoulme indeed this may be a problem, please check something on at least one of the returned value, for example is not an error.

Reply:
Thanks for review! I have updated to check the returned values don't have errors. I have edited the comment above with new performance benchmark numbers.